sections/delegate/LeftCol.tsx

```diff
@@ -0,0 +1,259 @@
+import { useState, useMemo, useEffect, FC, useCallback } from 'react';
+import { ethers } from 'ethers';
+import { useRecoilValue } from 'recoil';
+import { useRouter } from 'next/router';
+import styled from 'styled-components';
+import { Trans, useTranslation } from 'react-i18next';
+import Button from 'components/Button';
+import StructuredTab from 'components/StructuredTab';
+import Connector from 'containers/Connector';
+import Notify from 'containers/Notify';
+import {
+	ModalItemTitle as TxModalItemTitle,
+	ModalItemText as TxModalItemText,
+	NoTextTransform,
+	IconButton,
+	FlexDivRowCentered,
+} from 'styles/common';
+import { Svg } from 'react-optimized-image';
+import NavigationBack from 'assets/svg/app/navigation-back.svg';
+import GasSelector from 'components/GasSelector';
+import { LEFT_COL_WIDTH } from 'sections/delegate/constants';
+import { isWalletConnectedState } from 'store/wallet';
+import { Action, APPROVE_CONTRACT_METHODS } from 'queries/delegate/types';
+import {
+	FormContainer,
+	InputsContainer,
+	SettingsContainer,
+	SettingContainer,
+	ErrorMessage,
+	TxModalItem,
+} from 'sections/delegate/common';
+import { tx, getGasEstimateForTransaction } from 'utils/transactions';
+import {
+	normalizeGasLimit as getNormalizedGasLimit,
+	normalizedGasPrice as getNormalizedGasPrice,
+} from 'utils/network';
+import TxConfirmationModal from 'sections/shared/modals/TxConfirmationModal';
+import { useDelegates } from 'sections/delegate/contexts/delegates';
+import ActionSelector from './ActionSelector';
+
+const LeftCol: FC = () => {
+	const { t } = useTranslation();
+
+	const tabData = useMemo(
+		() => [
+			{
+				title: t('delegate.form.title'),
+				tabChildren: <Tab />,
+				key: 'main',
+				blue: true,
+			},
+		],
+		[t]
+	);
+
+	return (
+		<StructuredTab
+			singleTab={true}
+			boxPadding={20}
+			boxWidth={LEFT_COL_WIDTH}
+			tabData={tabData}
+			setPanelType={() => null}
+		/>
+	);
+};
+
+const Tab: FC = () => {
+	const { t } = useTranslation();
+	const router = useRouter();
+	const { connectWallet } = Connector.useContainer();
+	const isWalletConnected = useRecoilValue(isWalletConnectedState);
+	const { delegateApprovalsContract } = useDelegates();
+	const { monitorHash } = Notify.useContainer();
+
+	const [action, setAction] = useState<string>(Action.APPROVE_ALL);
+
+	const [gasPrice, setGasPrice] = useState<number>(0);
+	const [gasLimit, setGasLimitEstimate] = useState<number | null>(null);
+	const [delegateAddress, setDelegateAddress] = useState<string>('');
+
+	const [error, setError] = useState<string | null>(null);
+	const [txModalOpen, setTxModalOpen] = useState<boolean>(false);
+	const [buttonState, setButtonState] = useState<string | null>(null);
+
+	const shortenedDelegateAddress = useMemo(
```

Comment:
You could use `truncateAddress` function (from `lib/string.ts`)

Reply:
Awesome 👍🏽